modules/lang-painless/src/test/java/org/elasticsearch/painless/ElvisTests.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless;
+
+import static java.util.Collections.singletonMap;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+/**
+ * Tests for the Elvis operator ({@code ?:}).
+ */
+public class ElvisTests extends ScriptTestCase {
+    public void testBasics() {
```

Comment:
Oh boy.... That seems like a fairly breaky decision from a groovy compatibility standpoint. I don't know nearly enough to grasp the depth of the complexity that doing java-style unboxing would cause though....

Specifically for `?:`, not supporting automatic unboxing of the lhs would make the default chains fairly weird. We could refuse to support unboxing of the lhs unless it is a null safe dereference where the end of the chain is an int. That is kind of the analysis that we'd have to do to skip the box/unbox that I describe in that TODO. I didn't want to do it as part of this change because I figured it was big enough as is though. And I figured the unboxing played into the concept of a "default" anyway.